newrelic/data_source_newrelic_service_level_alert_helper.go

```diff
@@ -0,0 +1,154 @@
+package newrelic
+
+import (
+	"context"
+	"math/rand"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const (
+	alertTypeField              = "alert_type"
+	sliGUIDField                = "sli_guid"
+	sloTargetField              = "slo_target"
+	sloPeriodField              = "slo_period"
+	customConsumptionField      = "custom_tolerated_budget_consumption"
+	customEvaluationPeriodField = "custom_evaluation_period"
+	consumptionField            = "tolerated_budget_consumption"
+	evaluationPeriodField       = "evaluation_period"
+	thresholdField              = "threshold"
+	nrqlField                   = "nrql"
+
+	custom              = "custom"
+	fastBurn            = "fast_burn"
+	fastBurnPeriod      = 60
+	fastBurnConsumption = 2
```

Comment:
Some comment regarding naming collisions. These seems like variables that could benefit from typing with an enum style variable.

```go
type serviceLevelAlertType string

var serviceLevelAlertTypes = struct {
	CUSTOM    serviceLevelAlertType
	FAST_BURN serviceLevelAlertType
}{
	CUSTOM:    "custom",
	FAST_BURN: "fast_burn",
}
```

```
// ...
"alert_type": {
	Type:         schema.TypeString,
	Required:     true,
	ValidateFunc: validation.StringInSlice([]string{
		string(serviceLevelAlertTypes.CUSTOM), 
		string(serviceLevelAlertTypes.FAST_BURN),
	}, true),
},

// ...
switch serviceLevelAlertType(alertType) {
case serviceLevelAlertTypes.CUSTOM:
  // ...
case serviceLevelAlertTypes.FAST_BURN:
  // ...
}
```

Thoughts on this idea?